### PR TITLE
feat(platofrm): Added online/offline status detection in the platform

### LINKED ISSUE #499: PLATFORM: No Visual Feedback for Internet Connection Status
### No Visual Feedback for Internet Connection Status

**Describe the bug**  
The platform currently lacks visual feedback to indicate the user's internet connection status. There is no notification or visual indicator when the user is offline, which may lead to confusion when requests are made without a stable connection. The platform should display a red bar above the interface indicating that the user is offline and prompting them to reconnect before making any further requests.

**To Reproduce**  
1. Access the platform application.
2. Turn off your internet connection.
3. Observe that the platform gives no feedback on the internet connection status.

**Expected behavior**  
- A **red bar** should appear at the top of the interface when the user is offline, notifying them that they need to reconnect to the internet before making further requests.
- A **toaster notification** should also appear to notify the user that they are not connected to the internet.
- Once the user reconnects to the internet, another **toaster notification** should inform them they are back online, and the interface should **reset and sync the platform** with its current state.

**Screenshots**  
Check this Video out!
https://www.loom.com/share/f085a88f7ef6476fb6360e4d8bb22ffc?sid=f6aa37d2-612f-40f7-aae9-ca94579f39ca

**Desktop (please complete the following information):**
- OS: macOS Sonoma
- Browser: Chrome
- Version: -

**Smartphone (please complete the following information):**  
NA

**Additional context**  
This feature would prevent users from attempting actions while disconnected and ensure the platform automatically updates once the internet connection is restored.

### PR DESCRIPTION
**Pull request by Allan2000-Git**
## Description
This PR introduces functionality to detect and display the online/offline status of the platform using a toast message.

Fixes #499 

## Dependencies
N/A

## Future Improvements
N/A

## Mentions
@rajdip-b 

## Screenshots of relevant screens
I have throttled the network for testing the functionality.

https://www.loom.com/share/10eb3fcf518b40179a4ce52f92b46fa7

## Developer's checklist
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**
- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [x] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update
- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement

___

### **Description**
- Introduced `OnlineStatusHandler` component to manage online/offline status detection.
- Implemented `useOnlineStatus` hook to listen for network status changes and display toast notifications.
- Integrated the `OnlineStatusHandler` into the main application layout to provide real-time feedback on internet connectivity.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>Integrate online status handler in the application layout</code></dd></summary>
<hr>

apps/platform/src/app/layout.tsx

<li>Added <code>OnlineStatusHandler</code> component to the layout.<br> <li> Ensured <code>OnlineStatusHandler</code> is wrapped by <code>JotaiProvider</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/585/files#diff-27a60261e0be610328f75265d77101766eb077cf903f119a00e4eaea8db6139f">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>online-status-handler.tsx</strong><dd><code>Create OnlineStatusHandler component for status detection</code></dd></summary>
<hr>

apps/platform/src/components/common/online-status-handler.tsx

<li>Created <code>OnlineStatusHandler</code> component.<br> <li> Utilized <code>useOnlineStatus</code> hook within the component.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/585/files#diff-92c48b7a4ecbc9c7e0cb93f039cfa8c5a0e4f67392b0984a8cc93b4de98c4df0">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>use-online-status.ts</strong><dd><code>Implement useOnlineStatus hook for connection status</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/hooks/use-online-status.ts

<li>Implemented <code>useOnlineStatus</code> hook.<br> <li> Added event listeners for online and offline status.<br> <li> Displayed toast notifications for status changes.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/585/files#diff-15938f822bcaaf7dd009bfbb41f507224afce7e695679c5a03629898162b4325">+25/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by rajdip-b** [comment]
@kriptonian1 @poswalsameer can you drop your reviews?

**Comment by Allan2000-Git** [comment]
@rajdip-b I can implement suggestions given my code bot reviewer.

**Comment by rajdip-b** [comment]
Just the one that I marked, the others are not needed

> @rajdip-b I can implement suggestions given my code bot reviewer.

**Comment by poswalsameer** [comment]
> @kriptonian1 @poswalsameer can you drop your reviews?

Yeah sure1

**Comment by poswalsameer** [comment]
Hi @Allan2000-Git, can you please attach some screenshots with the PR?

**Comment by poswalsameer** [comment]
> @kriptonian1 @poswalsameer can you drop your reviews?

All looks good to me. I think instead of using setTimeout, we should have a `boolean` state to check the online/offline status of the user. And as this state changes, we can re-render the page using `useEffect` and show the corresponding toast message.

**Comment by Allan2000-Git** [comment]
> > @kriptonian1 @poswalsameer can you drop your reviews?
> 
> All looks good to me. I think instead of using setTimeout, we should have a `boolean` state to check the online/offline status of the user. And as this state changes, we can re-render the page using `useEffect` and show the corresponding toast message.

Cool, will make the change & update

File: apps/platform/src/hooks/use-online-status.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useEffect } from "react"
+import { toast } from "sonner";
+
+const statusHandler = () => {
+    if (navigator.onLine) {
+        toast.success("You are back online! Refreshing...");
+        setTimeout(() => {
+            window.location.reload();
+        }, 1000);
+    } else {
```
[the remark this answers is not in the document]
Comment (Allan2000-Git) [reply apps/platform/src/hooks/use-online-status.ts]: done

**Comment by poswalsameer** [comment]
> > > @kriptonian1 @poswalsameer can you drop your reviews?
> > 
> > 
> > All looks good to me. I think instead of using setTimeout, we should have a `boolean` state to check the online/offline status of the user. And as this state changes, we can re-render the page using `useEffect` and show the corresponding toast message.
> 
> Cool, will make the change & update

did the state thing worked?

**Comment by Allan2000-Git** [comment]
> > > > @kriptonian1 @poswalsameer can you drop your reviews?
> > > 
> > > 
> > > All looks good to me. I think instead of using setTimeout, we should have a `boolean` state to check the online/offline status of the user. And as this state changes, we can re-render the page using `useEffect` and show the corresponding toast message.
> > 
> > 
> > Cool, will make the change & update
> 
> did the state thing worked?

it's weird actually. it caused infinite re-render when i used state. also went through code, to keep it much cleaner, we do not require state actually as we're directly using the hook in the client component. state usage would be much more useful if we're using that state anywhere

**Comment by rajdip-b** [comment]
@kriptonian1 final thoughts? if you can approve it, I'll merge it asap

**Review by kriptonian1 — APPROVED** [review]
LGTM

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**